### ISSUE TITLE
explicit-length-check triggers on unrelated code when type is unknown

### ISSUE THREAD

**Issue by mkahvi** [sindresorhus/eslint-plugin-unicorn#2736]
The rule really wants to turn my `.size` into a `> 0` check when `.size` is actually a string, though in the code the rule is triggering it, the typing is missing (I am in regular JS where this is commonplace).

Here's minimal repro:
```js
class Foot {
  constructor(info) {
    this.info = info;
  }
  
  measure() {
    const base = this.info.size || "medium";
  }
}
```
I have much better defined data but it doesn't happen to be known where this invalid rule triggers.

**Comment by sindresorhus** [comment]
There is really nothing we can do about this without types.
